ls1intum/Artemis
# `Programming exercises`: improve user interface for programming exercise creation

### Checklist
#### General


#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


Resolves https://github.com/ls1intum/Artemis/issues/6326

### Description

Moves settings during programming exercise configuration.

As the list given in the PR was not comprehensive for all configuration items and not definitely agreed on, I did the following changes:

- Create new `Infrastructure` step
- Move preview repositories from `General Information` -> `Infrastructure`
- Move Checkboxes `Problem Statement` -> `Infrastructure`
  - Sequential Test Runs
  - Allow Offline IDE
  - Allow Online Editor
  - Show Test Names to Students
  - Publish Build Plan
  - Record Testwise Coverage

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Go to exercise creation
2. Create various exercises with different settings, switch both the normal as well as guided modes
3. Check that the exercise is created successfully with the correct settings after clicking the "finish" button for exercise creation 

### Review Progress







#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage





### Screenshots



Wizard mode
![localhost_9000_course-management_1_programming-exercises_new](https://github.com/ls1intum/Artemis/assets/25489463/8c5741c6-0d1a-492c-ac31-4fe664392b9f)


Normal mode
![localhost_9000_course-management_1_programming-exercises_new (1)](https://github.com/ls1intum/Artemis/assets/25489463/f40a1901-1479-442a-8e17-1549234b24d0)
